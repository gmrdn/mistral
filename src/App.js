import React from 'react'
import { useState, useEffect } from 'react';
import axios from 'axios';
import './App.css';
import Demain from './components/Demain.js'
import Prochain from './components/Prochain.js'

function isMistral(degrees) {
  console.log("is mistral")
  if (degrees > 280 && degrees < 360) {
    return true
  } else {
    return false
  }
}

function isStrong(speed) {
  if (speed > 5) {
    return true
  } else {
    return false
  }
}

function App() {
  const [windDirection, setDirection] = useState(0);
  const [windSpeed, setSpeed] = useState(0);
  const [mistral, setIsMistral] = useState(false);
  const [strong, setIsStrong] = useState(false);
  const [mistralBientot, setMistralBientot] = useState(false)
  const [prochainJour, setProchaiNjour] = useState("01/01/1970");

  useEffect(() => {
    getForecast()
  }, [])
  
  function getForecast() {
    axios.get('https://api.openweathermap.org/data/2.5/forecast?zip=13100,fr&APPID=f4e38397f380af9bab3b4dc73fab548a')
      .then(response => {
        setSpeed(response.data.list[0].wind.speed)
        setDirection(response.data.list[0].wind.deg)
        setIsMistral(isMistral(response.data.list[0].wind.deg))
        setIsStrong(isStrong(response.data.list[0].wind.speed))
      })
  }


  return (
    <>
    <h1 className="m-3">Gare au mistral !</h1>
    <div className="bg-light p-4">      
      <div className="container">  
        <div className="row">
          <div className="col py-md-5 mb-3">
            <Demain mistral={mistral} windDirection={windDirection} strong={strong} windSpeed={windSpeed} />
          </div>
          <div className="col py-md-5 mb-3">
            <Prochain />
          </div>
        </div> 
      </div>
    </div> 
    <footer className="d-flex justify-content-between bg-secondary p-3" id="mainFooter">
      Une app pour apprendre React et éviter d'être surpris par le mistral
    </footer>
    </>
  );

}

export default App