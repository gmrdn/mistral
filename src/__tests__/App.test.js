import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';


describe('Affichage page principale', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    expect(div).toMatchSnapshot();
  });


});
