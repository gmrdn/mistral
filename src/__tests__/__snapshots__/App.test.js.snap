// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Affichage page principale should match snapshot 1`] = `
<div>
  <h1
    class="m-3"
  >
    Gare au mistral !
  </h1>
  <div
    class="bg-light p-4"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col py-md-3 mb-3"
        >
          <div
            class="text-center"
          >
            <div
              class="spinner-border"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="col py-md-3 mb-3"
        >
          <div
            class="text-center"
          >
            <div
              class="spinner-border"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="d-flex justify-content-between bg-secondary p-3"
    id="mainFooter"
  >
    Une app pour apprendre React et éviter d'être surpris par le mistral
  </footer>
</div>
`;
